pyfarm/jobtypes/core/internals.py

```diff
@@ -774,20 +787,31 @@ def _remove_tempdirs(self):
         them from disk.  This work will be done in a thread so it does not
         block the reactor.
         """
-        reactor.callInThread(self._remove_directories, self._tempdirs)
+        assert isinstance(self._tempdirs, set)
+        if not self._tempdirs:
+            return
+
+        reactor.callInThread(self._remove_directories, self._tempdirs.copy())
+        self._tempdirs.clear()
 
     def _ensure_free_space_in_temp_dir(self, tempdir, space, minimum_age=None):
         """
         Ensures that at least space bytes of data can be stored on the volume
         on which tempdir is located, deleting files from tempdir if necessary.
 
-        WARNING: Will delete files in tempdir to reclaim storage space.
+        .. warning::
+
+            This will delete files in tempdir to reclaim storage space.
 
-        Will raise InsufficientSpaceError if enough space cannot be claimed.
+        :raises InsufficientSpaceError:
+            Raised if enough space cannot be claimed.
         """
+        assert isinstance(tempdir, STRING_TYPES)
+        assert isinstance(space, INTEGER_TYPES)
```

Comment:
Turns out, this was the problematic assert. The jobtype specified the required space as ```5e9```, which used to work just fine without this assert, but now fails because it's technically a float, not an int.

I suggest that, instead of the assert, you just convert the value to int and let this raise the appropriate exception if that fails.

Also, it would be nice if the asserts had an error message that tells us where the AssertionError came from when we catch it. Would make debugging a lot easier.